build-tools/packages/build-cli/src/commands/release/report-unreleased.ts

```diff
@@ -0,0 +1,229 @@
+						GITHUB_RELEASE_URL,
+						this.logger,
+					);
+					if (manifest !== undefined) {
+						await writeManifestToFile(devVersion, manifest, this.logger);
+					}
+				}
+			} else if (buildNumber === undefined) {
+				this.log(
+					`No successful build found for ${flags.sourceBranch} branch in the last 24 hours`,
+				);
+			}
+		} catch (error: unknown) {
+			this.errorLog(`Error creating manifest file: ${error}`);
+		}
+	}
+}
+
+/**
+ * Fetches the first successful build number in the last 24 hours.
+ * @returns The build number if successful, otherwise undefined.
+ */
+async function getFirstSuccessfulBuild(
+	authHeader: string,
+	adoUrl: string,
+	sourceBranch: string,
+	log?: Logger,
+): Promise<string | undefined> {
+	try {
+		const response = await fetch(adoUrl, { headers: { Authorization: authHeader } });
+		const data = await response.json();
+
+		const twentyFourHoursAgo = new Date();
+		twentyFourHoursAgo.setHours(twentyFourHoursAgo.getHours() - 24);
+
+		const successfulBuilds = data.value.filter(
+			(build: IBuildDetails) =>
+				build.definition.name === "Build - client packages" &&
+				build.status === "completed" &&
+				build.result === "succeeded" &&
+				build.sourceBranch === `refs/heads/${sourceBranch}` &&
+				new Date(build.finishTime) >= twentyFourHoursAgo,
+		);
+
+		return successfulBuilds.length > 0
+			? (successfulBuilds[0].buildNumber as string)
+			: undefined;
+	} catch (error) {
+		log?.errorLog("Error fetching successful builds:", error);
+		return undefined;
+	}
+}
+
+/**
+ * Fetches the dev version number released in the build feed.
+ * @param buildNumber - The build number.
+ * @returns The dev version number if found, otherwise undefined.
+ */
+async function fetchDevVersionNumber(
+	authHeader: string,
+	registryUrl: string,
+	packageName: string,
+	buildNumber: string,
+	log?: Logger,
+): Promise<string | undefined> {
+	try {
+		const response = await fetch(`${registryUrl}/${packageName}`, {
+			headers: { Authorization: authHeader },
+		});
+		const data = await response.json();
+		const buildVersionKey = Object.keys(data.time).find((key) => key.includes(buildNumber));
+
+		if (buildVersionKey !== undefined) {
+			return buildVersionKey;
+		}
+
+		log?.log(`No version with build number ${buildNumber} found.`);
+		return undefined;
+	} catch (error: unknown) {
+		log?.errorLog("Error fetching dev version number:", error);
+		return undefined;
+	}
+}
+
+/**
+ * Generates a modified manifest file with the specified version number.
+ * @param version - The version number.
+ */
+async function generateManifestObjectForDevReleases(
+	gitHubUrl: string,
+	log?: Logger,
+): Promise<PackageCaretRange | undefined> {
+	try {
+		const releasesResponse = await fetch(gitHubUrl);
+		const releases = await releasesResponse.json();
+
+		let manifestAsset;
+
+		for (const asset of releases[0].assets) {
+			const includesCaretJson: boolean = asset.name.includes(".caret.json");
```

Comment:
There's no need to use the caret one since you're replacing the values anyway. Either use the simple one as input or the full one (read it as a ReleaseReport).

Reply:
Per the other comment thread, this is OK. It's simpler than regenerating caret ranges for some packages, etc.